### PR TITLE
Fixed #25743 -- Bail early if the input to localize(_input) is a string

### PR DESCRIPTION
**Pull request by jaap3**
More details at https://code.djangoproject.com/ticket/25743

### CODE REVIEW

File: django/utils/formats.py [hunk 1]
```diff
@@ -172,18 +172,19 @@ def localize(value, use_l10n=None):
     If use_l10n is provided and is not None, that will force the value to
     be localized (or not), overriding the value of settings.USE_L10N.
     """
+    if isinstance(value, six.string_types):
+        return value
     if isinstance(value, bool):
         return mark_safe(six.text_type(value))
-    elif isinstance(value, (decimal.Decimal, float) + six.integer_types):
```
Comment (timgraham) [inline django/utils/formats.py]: I think there's an argument to using `elif` -- it makes it very explicit that the branches are mutually exclusive. Did I miss something or do you have a different argument?

Comment (jaap3) [inline django/utils/formats.py]: I'm not aware on any format rules/best practices around using if/elif in cases like this. It's my personal preference to only use elif when it is actually necessary for the logic to work correctly. (Which is not the case in this instance). I'll revert to using elif if that's more in line with the rest of Django.

Comment (timgraham) [inline django/utils/formats.py]: I'd keep it as-is then. It's confusing to combine cosmetic cleanups like that with optimizations anyway.

argument I read: http://stackoverflow.com/a/22783232/5112

File: django/utils/formats.py [hunk 2]
```diff
@@ -182,8 +184,7 @@ def localize(value, use_l10n=None):
         return date_format(value, use_l10n=use_l10n)
     elif isinstance(value, datetime.time):
         return time_format(value, 'TIME_FORMAT', use_l10n=use_l10n)
-    else:
-        return value
+    return value
```
Comment (jaap3) [inline django/utils/formats.py]: This is now consistent with localize_input

**Comment by timgraham** [comment]
merged in 9a2aca60304dc2e98f9ef45636e129d225cb981f, thanks!

**Pull request closed without merging**